applications/serial_lte_modem/doc/TWI_AT_commands.rst

```diff
@@ -0,0 +1,253 @@
+
+Read from TWI slave device #XTWIR
+=================================
+
+The ``#XTWIR`` command reads data from a TWI slave device.
+
+Set command
+-----------
+
+The set command allows you to read data from a TWI slave device.
+
+Syntax
+~~~~~~
+
+::
+
+   #XTWIR=<index>,<dev_addr>,<num_read>
+
+* The ``<index>`` parameter accepts the following integer values:
+
+  * ``0`` - Use TWI0 (``i2c0``).
+  * ``1`` - Use TWI1 (``i2c1``).
+  * ``2`` - Use TWI2 (``i2c2``).
+  * ``3`` - Use TWI3 (``i2c3``).
+
+  See :ref:`zephyr:nordic_nrf_twi`.
+
+* The ``<dev_addr>`` parameter is a hexadecimal string.
+  It represents the slave device address to read from.
+  The maximum length is 2 characters (for example, "DE" for 0xDE).
+* The ``<num_read>`` parameter is an unsigned 8-bit integer.
+  It represents the amount of data to read from the slave device.
+  The available range is from 0 to 255 bytes.
+
+Response syntax
+~~~~~~~~~~~~~~~
+
+::
+
+   #XTWIR: <size>
+   <data>
+
+* The ``<size>`` value is an unsigned integer.
+  It represents the actual number of bytes read from the slave device.
+* The ``<data>`` parameter is a hexadecimal string.
+  It represents the data read from the slave device.
+
+Example
+~~~~~~~
+
+::
+
+   AT#XTWIR=2,"76",1
+
+   #XTWIWR: 2
+   61
+   OK
+
+Read command
+------------
+
+The read command is not supported.
+
+Test command
+------------
+
+The test command is not supported.
+
+Write data and read from TWI device #XTWIWR
+===========================================
+
+The ``#XTWIWR`` command writes data to a TWI slave device and then reads data from the device.
+
+Set command
+-----------
+
+The set command allows you to first write data to a TWI slave device and then read that data back.
+
+Syntax
+~~~~~~
+
+::
+
+   #XTWIW=<index>,<dev_addr>,<data>,<num_read>
+
+* The ``<index>`` parameter accepts the following integer values:
+
+  * ``0`` - Use TWI0 (``i2c0``).
+  * ``1`` - Use TWI1 (``i2c1``).
+  * ``2`` - Use TWI2 (``i2c2``).
+  * ``3`` - Use TWI3 (``i2c3``).
+
+  See :ref:`zephyr:nordic_nrf_twi`.
+
+* The ``<dev_addr>`` parameter is a hexadecimal string.
+  It represents the slave device address to write to.
+  The maximum length is 2 characters (for example, "DE" for 0xDE).
+* The ``<data>`` parameter is a hexadecimal string.
+  It represents the data to be written to slave device.
+  The maximum length is 254 characters (for example, "DEADBEEF" for 0xDEADBEEF).
+* The ``<num_read>`` parameter is an unsigned 8-bit integer.
+  It represents the amount of data to read from the slave device.
+  The available range is from 0 to 255 bytes.
+
+Response syntax
+~~~~~~~~~~~~~~~
+
+::
+
+   #XTWIWR: <size>
+   <data>
+
+* The ``<size>`` value is an unsigned integer.
+  It represents the actual number of bytes read from the slave device.
+* The ``<data>`` parameter is a hexadecimal string.
+  It represents the data read from the slave device.
+
+Example
+~~~~~~~
+
+::
+
+   AT#XTWIWR=2,"76","D0",1
+
+   #XTWIWR: 2
+   61
```

Comment:
Is this the correct return value? I though it was hexadecimal (which might still be correct, I just don't understand how we get there from "D0". ;-) ).

Reply:
Yes, it (61) is correct return value for hexadecimal value 0x61.

AT#XTWIWR wraps the [i2c_write_read](https://docs.zephyrproject.org/apidoc/latest/group__i2c__interface_ga1273a9f8bdb002e0d6ece3a8c893a709.html) function that reading data from a register of a i2c slave device.
As the API description said it is a common operation "this is what I want", "now give it to me".

The example **AT#XTWIWR=2,"76","D0",1** reads the CHIP ID register of BME680 sensor on Thingy:91. The return value 61 is the CHIP ID of BME680.

As a reference, you can find the definition of registers here:
https://github.com/nrfconnect/sdk-zephyr/blob/master/drivers/sensor/bme680/bme680.h#L36
And the BME680 chip id definition:
https://github.com/nrfconnect/sdk-zephyr/blob/master/drivers/sensor/bme680/bme680.h#L14